src/VisualStudio/Core/Def/LanguageService/AbstractLanguageService`2.cs

```diff
@@ -82,32 +85,34 @@ public override IServiceProvider SystemServiceProvider
     /// <summary>
     /// Setup and TearDown go in reverse order.
     /// </summary>
-    internal void Setup()
+    public async Task SetupAsync(CancellationToken cancellationToken)
     {
-        this.ComAggregate = CreateComAggregate();
-
         // First, acquire any services we need throughout our lifetime.
-        this.GetServices();
+        // This method should only contain calls to acquire services off of the component model
+        // or service providers.  Anything else which is more complicated should go in Initialize
+        // instead.
+        this.EditorOptionsService = this.Package.ComponentModel.GetService<EditorOptionsService>();
+        this.Workspace = this.Package.ComponentModel.GetService<VisualStudioWorkspaceImpl>();
+        this.EditorAdaptersFactoryService = this.Package.ComponentModel.GetService<IVsEditorAdaptersFactoryService>();
+        this.AnalyzerFileWatcherService = this.Package.ComponentModel.GetService<AnalyzerFileWatcherService>();
 
-        // TODO: Is the below access to component model required or can be removed?
-        _ = this.Package.ComponentModel;
+        this.LanguageDebugInfo = CreateLanguageDebugInfo();
 
-        // Start off a background task to prime some components we'll need for editing
-        VsTaskLibraryHelper.CreateAndStartTask(VsTaskLibraryHelper.ServiceInstance, VsTaskRunContext.BackgroundThread,
-            () => PrimeLanguageServiceComponentsOnBackground());
+        // Start off a background task to prime some components we'll need for editing.
+        Task.Run(() =>
+        {
+            var formatter = this.Workspace.Services.GetLanguageServices(RoslynLanguageName).GetService<ISyntaxFormattingService>();
+            formatter?.GetDefaultFormattingRules();
+        }, cancellationToken).Forget();
```

Comment:
moved to normal Task.Run here.  Could switch back to VsTaskLibraryHelper.  Need to understand the value of that type.

Reply:
Fire and forget operations need to be tracked